src/dev/impl/DevToys/ViewModels/Tools/EncodersDecoders/JwtDecoderEncoder/JwtEncoderControlViewModel.cs

```diff
@@ -0,0 +1,245 @@
+                QueueNewTokenJob();
+            }
+        }
+
+        internal bool HasIssuer
+        {
+            get => SettingsProvider.GetSetting(JwtDecoderEncoderSettings.HasIssuer);
+            set
+            {
+                SettingsProvider.SetSetting(JwtDecoderEncoderSettings.HasIssuer, value);
+                OnPropertyChanged();
+                QueueNewTokenJob();
+            }
+        }
+
+        internal bool HasDefaultTime
+        {
+            get => SettingsProvider.GetSetting(JwtDecoderEncoderSettings.HasDefaultTime);
+            set
+            {
+                SettingsProvider.SetSetting(JwtDecoderEncoderSettings.HasDefaultTime, value);
+                OnPropertyChanged();
+                QueueNewTokenJob();
+            }
+        }
+
+        internal bool HasError
+        {
+            get => _hasError;
+            set
+            {
+                SetProperty(ref _hasError, value);
+            }
+        }
+
+        internal int ExpireYear
+        {
+            get => SettingsProvider.GetSetting(JwtDecoderEncoderSettings.ExpireYear);
+            set
+            {
+                if (value < 1)
+                {
+                    return;
+                }
+                SettingsProvider.SetSetting(JwtDecoderEncoderSettings.ExpireYear, value);
+                OnPropertyChanged();
+                QueueNewTokenJob();
+            }
+        }
+
+        internal int ExpireMonth
+        {
+            get => SettingsProvider.GetSetting(JwtDecoderEncoderSettings.ExpireMonth);
+            set
+            {
+                if (value < 1)
+                {
+                    return;
+                }
+                SettingsProvider.SetSetting(JwtDecoderEncoderSettings.ExpireMonth, value);
+                OnPropertyChanged();
+                QueueNewTokenJob();
+            }
+        }
+
+        internal int ExpireDay
+        {
+            get => SettingsProvider.GetSetting(JwtDecoderEncoderSettings.ExpireDay);
+            set
+            {
+                if (value < 1)
+                {
+                    return;
+                }
+                SettingsProvider.SetSetting(JwtDecoderEncoderSettings.ExpireDay, value);
+                OnPropertyChanged();
+                QueueNewTokenJob();
+            }
+
+        }
+
+        internal int ExpireHour
+        {
+            get => SettingsProvider.GetSetting(JwtDecoderEncoderSettings.ExpireHour);
+            set
+            {
+                if (value < 1)
+                {
+                    return;
+                }
+                SettingsProvider.SetSetting(JwtDecoderEncoderSettings.ExpireHour, value);
+                OnPropertyChanged();
+                QueueNewTokenJob();
+            }
+        }
+
+        internal int ExpireMinute
+        {
+            get => SettingsProvider.GetSetting(JwtDecoderEncoderSettings.ExpireMinute);
+            set
+            {
+                if (value < 1)
```

Comment:
I'm a little confused by this. In the UI, it looks like the default value in `Expire in minutes(s)` is `0`, yet, 0 should be forbidden according to this logic, isn't it?